htdocs/core/db/mysqli.class.php

```diff
@@ -939,7 +939,7 @@ public function DDLDropField($table, $field_name)
 	public function DDLCreateUser($dolibarr_main_db_host, $dolibarr_main_db_user, $dolibarr_main_db_pass, $dolibarr_main_db_name)
 	{
 		// phpcs:enable
-		$sql = "CREATE USER '".$this->escape($dolibarr_main_db_user)."'";
+		$sql = "CREATE USER '".$this->escape($dolibarr_main_db_user)."'@'".$this->escape($dolibarr_main_db_host)."' IDENTIFIED BY '".$this->escape($dolibarr_main_db_pass)."'";
```

Comment:
The dolibarr_main_db_host is the host of the web server. In some case, the database mayy be into another one.
So i think we should not add the '@'".$this->escape($dolibarr_main_db_host)."' here, only add the IDENTIFIED BY so database will use the default value for domain name (like actually).